tests/integration/security/pass_through_filter_chain_test.go

```diff
@@ -52,206 +49,182 @@ func TestPassThroughFilterChain(t *testing.T) {
-				},
-			}
+					// For workload c, there is only authN policy that enables mTLS (Strict).
+					// The request should be denied because the x is always using plain text.
+					{
+						target: cSet[0],
+						port:   8085,
+						schema: protocol.HTTP,
+						want:   false,
+					},
+					{
+						target: cSet[0],
+						port:   8086,
+						schema: protocol.HTTP,
+						want:   false,
+					},
+					{
+						target: cSet[0],
+						port:   8087,
+						schema: protocol.TCP,
+						want:   false,
+					},
+					{
+						target: cSet[0],
+						port:   8088,
+						schema: protocol.TCP,
+						want:   false,
+					},
 
-			for _, tc := range cases {
-				name := fmt.Sprintf("x->%s:%d[%t]", tc.target.Config().Service, tc.port, tc.want)
-				from := getWorkload(a, t)
-				// The request should be handled by the pass through filter chain.
-				host := fmt.Sprintf("%s:%d", getWorkload(tc.target, t).Address(), tc.port)
-				request := &epb.ForwardEchoRequest{
-					Url:     fmt.Sprintf("%s://%s", tc.schema, host),
-					Count:   1,
-					Message: "HelloWorld",
-					Headers: []*epb.Header{
-						{
-							Key:   "Host",
-							Value: host,
-						},
+					// For workload d, there is only authN policy that enables mTLS (Permissive).
+					// The request should be allowed because the x is always using plain text.
+					{
+						target: dSet[0],
+						port:   8085,
+						schema: protocol.HTTP,
+						want:   true,
+					},
+					{
+						target: dSet[0],
+						port:   8086,
+						schema: protocol.HTTP,
+						want:   true,
+					},
+					{
+						target: dSet[0],
+						port:   8087,
+						schema: protocol.TCP,
+						want:   true,
+					},
+					{
+						target: dSet[0],
+						port:   8088,
+						schema: protocol.TCP,
+						want:   true,
 					},
 				}
-				t.Run(name, func(t *testing.T) {
-					retry.UntilSuccessOrFail(t, func() error {
-						responses, err := from.ForwardEcho(context.TODO(), request)
-						if tc.want {
-							if err != nil {
-								return fmt.Errorf("want allow but got error: %v", err)
-							}
-							if len(responses) < 1 {
-								return fmt.Errorf("received no responses from request to %s", host)
-							}
-							if tc.schema == protocol.HTTP && response.StatusCodeOK != responses[0].Code {
-								return fmt.Errorf("want status %s but got %s", response.StatusCodeOK, responses[0].Code)
-							}
-						} else {
-							// Check HTTP forbidden response
-							if len(responses) >= 1 && response.StatusCodeForbidden == responses[0].Code {
-								return nil
-							}
-
-							if err == nil || !strings.Contains(err.Error(), "EOF") {
-								return fmt.Errorf("want error EOF but got: %v", err)
-							}
+				ctx.NewSubTest(fmt.Sprintf("In %s", cluster.Name())).Run(func(ctx framework.TestContext) {
+					for _, tc := range cases {
+						name := fmt.Sprintf("x->%s:%d[%t]", tc.target.Config().Service, tc.port, tc.want)
+						a := apps.A.Match(echo.InCluster(cluster)).Match(echo.Namespace(ns.Name()))[0]
```

Comment:
Seems like where you do matching for `aSet` above, you'll only be getting the one instance.

Reply:
Changed to GetOrFail